### PR TITLE
Adds .XOR. operator for logicals to IntrTypes

### PR DESCRIPTION
**Pull request by aarograh**

### CODE REVIEW

**Comment by djabaay** [comment]
As an fyi, this intrinsic already exists in Fortran as .NEQV. which is the updated standard for .XOR.  

.XOR. is for f77 compatibility.

https://gcc.gnu.org/onlinedocs/gfortran/XOR.html

File: src/IntrType.f90 [hunk 1]
```diff
@@ -1000,6 +1008,19 @@ ELEMENTAL FUNCTION notequalto_logical(l1,l2) RESULT(bool)
     ENDFUNCTION notequalto_logical
 !
```
Comment (djabaay) [inline src/IntrType.f90]: xor_logical is the same function as notequalto_logical.

Comment (aarograh) [reply src/IntrType.f90]: Ah, good call... totally didn't know this was in there.  Do we want to revert the change, or just leave both?

**Comment by aarograh** [comment]
Dan, thanks for clarifying.  I didn't realize the .NEQV. operator existed.  I've always checked for .XOR. (since that is the standard name for this operator in other languages) with no luck.

If we want to revert this change I can just switch to using .NEQV. or the overloaded /= operators where appropriate.

**Comment by djabaay** [comment]
Yea, I know the feeling, I've looked for it in the past as well since that's what I was familiar with.  I'd say revert the changes, since we can use the /= operator which is a little shorter and cleaner.

**Pull request merged**